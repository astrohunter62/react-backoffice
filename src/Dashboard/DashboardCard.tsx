import React from "react";
import classNames from "classnames";
import {
  Card,
  CardContent,
  Typography,
  withStyles,
  Avatar,
} from "@material-ui/core";
import DisabledIcon from "@material-ui/icons/Lock";

const styles = (theme: any) => ({
  root: {
    cursor: "pointer",
  },
  content: {
    paddingBottom: `${theme.spacing(2)}px !important`,
  },
  avatar: {
    float: "left",
    marginRight: theme.spacing(2),
    backgroundColor: theme.palette.secondary.dark,
  },
  title: {
    marginBottom: 0,
  },
  disabled: {
    opacity: 0.75,
    pointerEvents: "none",
    filter: "grayscale(30%)",
  },
  icon: {
    float: "right",
    opacity: 0.5,
  },
});

type DashboardCardProps = {
  title: string;
  description?: string;
  handleClick: (...args: any[]) => any;
  icon?: (...args: any[]) => any;
  isDisabled?: boolean;
  classes: {
    [key: string]: string;
  };
};

const DashboardCard: React.SFC<DashboardCardProps> = ({
  title,
  description,
  handleClick,
  icon: Icon,
  isDisabled,
  classes,
}) => {
  const rootClasses = classNames({
    [classes.root]: true,
    [classes.disabled]: isDisabled,
  });

  return (
    <Card onClick={handleClick} className={rootClasses}>
      <CardContent className={classes.content}>
        {isDisabled ? <DisabledIcon className={classes.icon} /> : null}

        {Icon ? (
          <Avatar className={classes.avatar}>
            <Icon />
          </Avatar>
        ) : null}

        <Typography variant="h5" className={classes.title} component="h2">
          {title}
        </Typography>

        {description ? (
          <Typography variant="body2">{description}</Typography>
        ) : null}
      </CardContent>
    </Card>
  );
};
DashboardCard.defaultProps = {
  isDisabled: false,
};

export default withStyles(styles as any)(DashboardCard);
