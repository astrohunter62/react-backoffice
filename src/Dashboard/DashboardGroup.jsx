import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'

import Grid from 'material-ui/Grid'
import Typography from 'material-ui/Typography'

import DashboardCard from './DashboardCard'

const styles = theme => ({
  headline: {
    marginTop: theme.spacing.unit * 3,
    marginBottom: theme.spacing.unit * 2,
  },
})

const DashboardGroup = ({
  title,
  cards,
  onClick,
  classes,
}) => (
  <div>
    <div className={classes.headline}>
      <Typography variant="headline">
        {title}
      </Typography>
    </div>

    <Grid container spacing={16}>

      {cards ? cards.map(card => (
        <Grid item xs={12} sm={4} key={`group-card-${card.id}`}>
          <DashboardCard
            title={card.title}
            description={card.description}
            handleClick={() => onClick(card.link)}
            icon={card.icon}
            disabled={card.disabled}
          />
        </Grid>
      )) : null}

    </Grid>
  </div>
)

DashboardGroup.propTypes = {
  title: PropTypes.string.isRequired,
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(DashboardGroup)
