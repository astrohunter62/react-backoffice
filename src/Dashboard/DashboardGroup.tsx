import React from "react";
import { Grid, Typography, withStyles } from "@material-ui/core";
import DashboardCard from "./DashboardCard";
const styles = (theme: any) => ({
  headline: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(2),
  },
});

type DashboardGroupProps = {
  title: string;
  cards: Record<string, any>[];
  onClick: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};
const DashboardGroup: React.SFC<DashboardGroupProps> = ({
  title,
  cards,
  onClick,
  classes,
}) => (
  <div>
    <div className={classes.headline}>
      <Typography variant="h5">{title}</Typography>
    </div>

    <Grid container spacing={2}>
      {cards
        ? cards.map((card) => (
            <Grid item xs={12} sm={4} key={`group-card-${card.id}`}>
              <DashboardCard
                title={card.title}
                description={card.description}
                handleClick={() => onClick(card.link)}
                icon={card.icon}
                isDisabled={card.isDisabled}
              />
            </Grid>
          ))
        : null}
    </Grid>
  </div>
);

export default withStyles(styles)(DashboardGroup);
