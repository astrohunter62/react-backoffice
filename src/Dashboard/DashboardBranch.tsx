import React from "react";
import { Typography, withStyles } from "@material-ui/core";
import DashboardGroup from "./DashboardGroup";
const styles = (theme: any) => ({
  root: {
    flexGrow: 1,
    maxWidth: theme.spacing(140),
    marginLeft: "auto",
    marginRight: "auto",
  },
  headline: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(2),
  },
});

interface Groups {
  id: string;
  title: string;
  cards: any;
}

type DashboardBranchProps = {
  data: {
    groups: Groups[];
    [key: string]: any;
  };
  onClick: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};

const DashboardBranch: React.SFC<DashboardBranchProps> = ({
  data,
  onClick,
  classes,
}) => (
  <div className={classes.root}>
    <div className={classes.headline}>
      <Typography variant="h3">{data.title}</Typography>
      <Typography variant="body2">{data.description}</Typography>
    </div>

    {data.groups
      ? data.groups.map((group) => (
          <DashboardGroup
            key={`group-${group.id}`}
            onClick={onClick}
            {...group}
          />
        ))
      : null}
  </div>
);
export default withStyles(styles)(DashboardBranch);
