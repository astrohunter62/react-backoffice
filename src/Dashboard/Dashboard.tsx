import React from "react";
import { withRouter } from "react-router-dom";
import DashboardBranch from "./DashboardBranch";
type WithDashboardProps = {
  history: {
    [key: string]: any;
  };
};
const withDashboard = (Component: any) =>
  class WithDashboard extends React.Component<WithDashboardProps, any> {
    constructor(props: WithDashboardProps) {
      super(props);

      this.handleClick = this.handleClick.bind(this);
    }
    handleClick(url: string) {
      this.props.history.push(url);
    }
    render() {
      return <Component {...this.props} onClick={this.handleClick} />;
    }
  };
const Dashboard = withDashboard(DashboardBranch);
export default withRouter(Dashboard as any) as any;
