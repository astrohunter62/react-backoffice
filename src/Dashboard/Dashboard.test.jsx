import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Dashboard from './'

it('renders correctly', () => {
  const tree = renderer
    .create((
      <Router>
        <Dashboard
          data={{
            title: 'Backoffice',
            description: 'Visual Testing for Backoffice Framework',
            groups: [{
              title: 'Title',
              cards: [{
                title: 'title',
                description: 'desc',
              }],
            }],
          }}
        />
      </Router>
    ))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
