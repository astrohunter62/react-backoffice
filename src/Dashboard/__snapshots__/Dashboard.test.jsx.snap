// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(_class)
    data={
      Object {
        "description": "Visual Testing for Backoffice Framework",
        "groups": Array [
          Object {
            "cards": Array [
              Object {
                "description": "desc",
                "title": "title",
              },
            ],
            "title": "Title",
          },
        ],
        "title": "Backoffice",
      }
    }
  />
</Router>
`;
